var/spack/repos/builtin/packages/flecsi/package.py

```diff
@@ -46,8 +46,8 @@ class Flecsi(CMakePackage, CudaPackage, ROCmPackage):
         description="Set Caliper Profiling Detail",
         multi=False,
     )
-    variant("kokkos", default=False, description="Enable Kokkos Support")
-    variant("openmp", default=False, description="Enable OpenMP Support")
+    variant("kokkos", default=False, description="Enable Kokkos Support", when="@:2.3.1")
+    variant("openmp", default=False, description="Enable OpenMP Support", when="@:2.3.1")
```

Comment:
Do these variants still _work_ if you're selecting an earlier version?  The changes below look like they wouldn't.

Reply:
Because these versions still have CMake options that control whether Kokkos and OpenMP are enabled. If we decide to remove CudaPackage/ROCmPackage we can't do it selectively for some versions. We either do it completely or we must leave it.

Reply:
Rolling those changes back for now and only adding the release.

Reply:
@opensdh Keeping the cuda and rocm variants for now. This keeps the Spack specs stable for the older releases. The additional rules now support `flecsi ^kokkos +cuda cuda_arch=X` and make sure they propagate to ourselves. So you can now either specify it for FleCSI *or* inherit it from Kokkos.

Reply:
What I meant about getting rid of `+kokkos` wasn't that the previous versions didn't need the _information_, but that they could key off `^kokkos`, but now I realize that `sim ^flecsi ^kokkos` doesn't indicate whether or not FleCSI should consider Kokkos as _its_ dependency.